### PR TITLE
Api promises

### PR DESCRIPTION
**Pull request by oscarwang20**
### Summary 

This pull request addresses the mishandling of promises caught by this [tslint rule](https://www.notion.so/cornelldti/API-Layer-Fix-mishandling-of-promises-ae4235e0935d41ccada327a5a5cf7b97?pvs=4), namely `"checksConditionals"` and `"checksSpreads"`. 

### Notion/Figma Link 

https://www.notion.so/cornelldti/API-Layer-Fix-mishandling-of-promises-ae4235e0935d41ccada327a5a5cf7b97?pvs=4

### Test Plan 

Everything looks good after TSLinting

### Notes 

In `api.ts`, the rule is disabled because the `"chekcsVoidReturn"` rule is violated on six instances: 

image

I tried to fix this by following the TSLint documentation for correct and incorrect code, and with some help from [this post](https://stackoverflow.com/questions/65670264/typescript-node-express-router-and-second-argument-typescript-eslint-no-misused), but neither seems to work. I also found further documentation of this issue on [this thread](https://github.com/expressjs/express/issues/4892). Everything on the site still works the same as before.

### CODE REVIEW

File: backend/src/api.ts [hunk 1]
```diff
@@ -187,7 +188,8 @@ router.get('/allApprovedMembers', async (_, res) => {
   res.status(200).json({ members });
 });
 router.get('/membersFromAllSemesters', async (_, res) => {
-  res.status(200).json(await MembersDao.getMembersFromAllSemesters());
+  const members = await MembersDao.getMembersFromAllSemesters();
+  res.status(200).json({ members });
 });
```
Comment (henry-li-06) [inline backend/src/api.ts:193]: Don't need to split this up into 2 lines

File: backend/scripts/update-dev-db.ts [hunk 2]
```diff
@@ -34,8 +34,16 @@ const rewriteDbData = async (devDb: FirebaseFirestore.Firestore, data: DbData):
     .then((collections) => collections.map((collection) => collection.id));
 
   await Promise.all(allCollections.map((collection) => deleteCollection(devDb, collection, 10)));
-  return data.map((collection) =>
-    collection.docs.forEach((doc) => devDb.doc(`${collection.id}/${doc.id}`).set(doc.data))
+  return Promise.all(
```
Comment (andrew032011) [inline backend/scripts/update-dev-db.ts:37]: Can you await this outer `Promise.all` as well?

**Review by andrew032011 — APPROVED** [review]
Minor comment. Thanks for doing this!

File: backend/src/API/devPortfolioAPI.ts [hunk 3]
```diff
@@ -40,8 +40,8 @@ export const createNewDevPortfolio = async (
   instance: DevPortfolio,
   user: IdolMember
 ): Promise<DevPortfolio> => {
-  const canCreateDevPortfolio = await PermissionsManager.isLeadOrAdmin(user);
```
Comment (andrew032011) [inline backend/src/API/devPortfolioAPI.ts:43]: Personally I don't see the need to change all these to `isLeadOrAdmin`. `canCreateDevPortfolio` is fine and is more descriptive. But I'll leave it up to you.

Comment (oscarwang20) [reply backend/src/API/devPortfolioAPI.ts:43]: I think the method names are descriptive enough, and changing it to `isLeadOrAdmin` keeps it consistent with the rest of the original file.

**Pull request merged**